design-proposals/eim-cli.md

```diff
@@ -0,0 +1,254 @@
+# Design Proposal: EIM CLI
+
+Author(s): damiankopyto
+
+Last updated: 01/05/2025
+
+## Abstract
+
+The Edge Orchestrator requires a management via a CLI tool, an extensible EMF CLI will be designed/implemented as per the Orchestrator CLI ADR (https://github.com/open-edge-platform/edge-manageability-framework/pull/246/files).
+As part of this CLI a number of EIM specific actions/features/workflows should be supported.
+This design proposal will focus on the details required for the EIM support.
+
+## Proposal
+
+This document proposes to include the EIM related functionality as subset of commands of the EMF CLI. The workflows called out in the following section are to be supported and represented through commands of the EMF CLI, the usability and functionality of the commands should be on par with the UI usage. The CLI will call the necessary APIs in order to execute the actions.
+
+## Workflows
+
+The following workflows are to be supported initially - the list may be expanded in the future if necessary functionality needs to be supported via CLI. It is assumed that the Bulk Import Tool [(BIT)](https://github.com/open-edge-platform/infra-core/tree/main/bulk-import-tools) will be integrated as part of the CLI effort.
+
+> **NOTE**: The commands and the order of commands at this stage is in no way final - they are just placeholders until we agree on actual command structure overall.
+
+### Ability to register/onboard/provision a host
+
+#### Local Options
+
+The local options associated with the `host` object/noun would help with filtering and providing arguments to the commands:
+
+- `-o` `registered`/`onboarded`/`provisioned`/`all`/`deauthorized`/`unknown`/`error` - helps filter out the output of `list` command/verb ( TODO maybe instead of `-o` for output change to `-s` for status???)
```

Comment:
If we want to keep the same convention of kubectl the it would make more sense to use `-o` for output formatting rather than filtering flags. Instead we could use `--registered`, `--onboarded` and so on to filter list output and keep `-o` for json, yaml or wide options.